webapi/src/test/scala/org/knora/webapi/routing/admin/ProjectsRouteZSpec.scala

```diff
@@ -38,99 +46,183 @@ object ProjectRouteZSpec extends ZIOSpecDefault {
+
+  private val expectedResponseNotFoundException: NotFoundException = NotFoundException("xxxx")
 
   // Expectations and layers for ActorToZioBridge mock
   private val commonLayers: URLayer[ActorToZioBridge, ProjectsRouteZ] =
     ZLayer.makeSome[ActorToZioBridge, ProjectsRouteZ](AppConfig.test, ProjectsRouteZ.layer, RestProjectsService.layer)
 
+  private val expectMessageToProjectResponderMultipleProjects: ULayer[ActorToZioBridge] =
+    ActorToZioBridgeMock.AskAppActor
+      .of[ProjectsGetResponseADM]
+      .apply(
+        Assertion.equalTo(expectedProjectsGetRequestADMSuccess),
+        Expectation.value(expectedResponseMultipleProjectsSuccess)
+      )
+      .toLayer
+
+  private val expectMessageToProjectResponderMultipleProjectsFailure: ULayer[ActorToZioBridge] =
+    ActorToZioBridgeMock.AskAppActor
+      .of[NotFoundException]
+      .apply(
+        Assertion.equalTo(expectedProjectsGetRequestADMFailure),
+        Expectation.failure(expectedResponseNotFoundException)
+      )
+      .toLayer
+
   private val expectMessageToProjectResponderIri: ULayer[ActorToZioBridge] =
     ActorToZioBridgeMock.AskAppActor
       .of[ProjectGetResponseADM]
-      .apply(Assertion.equalTo(expectedRequestSuccessIri), Expectation.value(expectedResponseSuccess))
+      .apply(
+        Assertion.equalTo(expectedProjectGetRequestADMSuccessIri),
+        Expectation.value(expectedResponseOneProjectSuccess)
+      )
+      .toLayer
+
+  private val expectMessageToProjectResponderFailure: ULayer[ActorToZioBridge] =
+    ActorToZioBridgeMock.AskAppActor
+      .of[NotFoundException]
+      .apply(
+        Assertion.equalTo(expectedProjectGetRequestADMFailureIri),
+        Expectation.failure(expectedResponseNotFoundException)
+      )
       .toLayer
 
   private val expectMessageToProjectResponderShortname: ULayer[ActorToZioBridge] =
     ActorToZioBridgeMock.AskAppActor
       .of[ProjectGetResponseADM]
-      .apply(Assertion.equalTo(expectedRequestSuccessShortname), Expectation.value(expectedResponseSuccess))
+      .apply(
+        Assertion.equalTo(expectedProjectGetRequestADMSuccessShortname),
+        Expectation.value(expectedResponseOneProjectSuccess)
+      )
       .toLayer
 
   private val expectMessageToProjectResponderShortcode: ULayer[ActorToZioBridge] =
     ActorToZioBridgeMock.AskAppActor
       .of[ProjectGetResponseADM]
-      .apply(Assertion.equalTo(expectedRequestSuccessShortcode), Expectation.value(expectedResponseSuccess))
+      .apply(
+        Assertion.equalTo(expectedProjectGetRequestADMSuccessShortcode),
+        Expectation.value(expectedResponseOneProjectSuccess)
+      )
       .toLayer
 
   private val expectNoInteractionWithProjectsResponderADM = ActorToZioBridgeMock.empty
 
   val spec: Spec[Any, Serializable] =
     suite("ProjectsRouteZSpec")(
+      suite("get all projects")(
+        test("given projects exist, return all projects") {
+          val urlGetAllProjects = URL.empty.setPath(basePathProjects)
```

Comment:
IMO it would be more readable, if the suites and tests were more spaced out - or if the suites were even in separate variables that are then just put together.  
Like this it looks very much like a "Tatzelwurm" to me

Reply:
I totally agree ;) I will separate them.